docs/source/howto/codes.rst

```diff
@@ -45,8 +45,90 @@ Design guidelines
 Running external codes
 ======================
 
-`#3987`_
+To run an external code with AiiDA, you will need to use an appropriate :ref:`calculation plugin <topics:plugins>` that knows how to transform the input nodes into the input files that the code expects, copy everything in the code's machine, run the calculation and retrieve the results.
+This plugin would have already been created, either by you (see section on :ref:`interfacing external codes <how-to:codes:plugin>`) or by other AiiDA plugin-developers (available packages are listed in the `plugin registry <https://aiidateam.github.io/aiida-registry/>`_), and has been properly installed in your environment.
+You can check which ones you have currently available:
 
+.. code-block:: bash
+
+    $ verdi plugin list aiida.calculations
+
+As an example, we will show how to use the ``arithmetic.add`` plugin, which is a pre-installed plugin implemented in `bash` to sum two integers.
+You can access it with the ``CalculationFactory``:
+
+.. code-block:: python
+
+    from aiida.plugins import CalculationFactory
+    calculation_class = CalculationFactory('arithmetic.add')
+
+Next, we provide the inputs for the code when running the calculation.
+Use ``verdi plugin`` to determine what inputs a specific plugin expects:
+
+.. code-block:: bash
+
+    $ verdi plugin list aiida.calculations arithmetic.add
+    (...)
+        Inputs:
+               code:  required  Code        The `Code` to use for this job.
+                  x:  required  Int, Float  The left operand.
+                  y:  required  Int, Float  The right operand.
+    (...)
+
+You will see that 3 inputs nodes are required: two containing the values to add up (``x``, ``y``) and one containing information about the specific code to execute (``code``).
+If you already have these nodes in your database, you can get them by :ref:`querying for them <how-to:data:finding-data>` or using ``orm.load_node(<PK>)``.
+Otherwise, you will need to create them as shown below (note that you `will` need to already have the ``localhost`` computer configured):
```

Comment:
I think we really need at a minimum a link to a How-To configure a computer. I don't think it exists yet, but you can already create the link reference and stub, and open the corresponding issue

Reply:
Should it be part of "How to manage your installation"?

Reply:
I would more be tempted to put it in this same How to, because really the only reason you need to configure a computer is to run `CalcJobs`. Not sure where, maybe after "Interfacing external codes" for now? Something like "Configuring a remote computer"

Reply:
["Configuring remote resources"](https://github.com/aiidateam/aiida-core/issues/4123) sounds good?

Reply:
I think "resources" is too broad in this case, because it does not necessarily convey that we are talking about computers. We could specify "Configuring remote computing resources" but in that case I think "Configuring remote computers" is better, as it is clearer and more concise, while saying the exact thing

Reply:
Ok, changed. Ready for final review now.